openfl/native/native.py

```diff
@@ -98,16 +98,16 @@ def update_plan(override_config):
     Returns:
         None
     """
-    plan_path = 'plan/plan.yaml'
-    flat_plan_config = get_plan(return_complete=True)
+    plan, flat_plan_config = get_plan(return_complete=True)
     for k, v in override_config.items():
         if k in flat_plan_config:
-            flat_plan_config[k] = v
             logger.info(f'Updating {k} to {v}... ')
         else:
-            logger.info(f'Key {k} not found in plan. Ignoring... ')
-    plan_config = unflatten(flat_plan_config, '.')
-    Plan.Dump(Path(plan_path), plan_config)
```

Comment:
@psfoley The fix for YAML loading is here. The solution is that we don't dump updated plan to disk but just keep it as an object in memory. So even if we use an updated plan in the experiment, `plan/plan.yaml` will always contain default (non-overridden) Plan settings. My concern here is that we will never get an updated version of the Plan during the `fx workspace export` of this workspace. Shall we support exporting of workspaces created for native API calls?